### PR TITLE
Adding tests for dex plus restructuring and formatting

### PR DESCRIPTION
**Pull request by tzabbi**
# Pull Request Template for Kubeflow manifests Issues

## ✅ Contributor checklist
  - Make sure you have tested with kustomize. See [Installation Prerequisites](https://github.com/kubeflow/manifests#prerequisites)
  - All the commits have been _signed-off_  (To pass the `DCO` check) ✅
  - Submit the [Contributor License Agreements](https://cla.developers.google.com/clas) (To pass the `cla/google` check) ✅

---     
 
> You can join the CNCF Slack and access our meetings at the [Kubeflow Community](https://www.kubeflow.org/docs/about/community/) website. Our channel on the CNCF Slack is here [**#kubeflow-platform**](https://app.slack.com/client/T08PSQ7BQ/C073W572LA2).

### CODE REVIEW

**Comment by juliusvonkohout** [comment]
@kromanow94 @kimwnasptd can you take a look at this. Somehow noone noticed it during distribution testing. If this is a problem, we need to have a test for the DEX login flow.

**Comment by juliusvonkohout** [comment]
@tzabbi can you extend the dex test, such that this does not happen again?

**Comment by tzabbi** [comment]
@juliusvonkohout not sure how to imporve the test. If I see it correctly your dex test is just to build and deploy the kubernetes manifests right? https://github.com/kubeflow/manifests/blob/master/.github/workflows/dex_test.yaml#L31

In this case we have to log into the system.

**Comment by tzabbi** [comment]
I will try a curl.

**Comment by juliusvonkohout** [comment]
Does it only apply the manifests so far? Yes.
After https://github.com/kubeflow/manifests/blob/ace875bbed9e284483c545cdc1c2cb60ad5b24a3/.github/workflows/dex_test.yaml#L31 you could add something similar to https://github.com/kubeflow/manifests/blob/ace875bbed9e284483c545cdc1c2cb60ad5b24a3/.github/workflows/pipeline_test.yaml#L57 that emulates a session login and gets the redirect URL. 

https://www.kubeflow.org/docs/components/pipelines/user-guides/core-functions/connect-api/#example-for-dex might have some information.

**Comment by kromanow94** [comment]
Hello all,

This wasn't an issue because the `example` kustomize correctly uses the `dex/overlays/oauth2-proxy` for dex.

Please see:
```
$ git log -1 
commit ace875bbed9e284483c545cdc1c2cb60ad5b24a3 (HEAD -> upstream.master, upstream/master)
Author:     biswajit-9776 <115724497+biswajit-9776@users.noreply.github.com>
AuthorDate: Tue Jul 23 19:21:02 2024 +0530
Commit:     GitHub <noreply@github.com>
CommitDate: Tue Jul 23 13:51:02 2024 +0000

    PSS labels for the profile controller (#2778)

$ grep Dex -A1 example/kustomization.yaml 
# Dex
- ../common/dex/overlays/oauth2-proxy

$ kustomize build example 2>/dev/null | grep redirectURIs
      redirectURIs: ["/oauth2/callback"]
```

But, I suppose with the changed default from the oidc-authservice to the oauth2-proxy, we can as well just delete the oidc-authservice from `common/oidc-client/oidc-authservice` (maybe even move `common/oidc-client/oauth2-proxy` to `common/oauth2-proxy`), and remove all references to the `oidc-authservice`, wherever they might be (this includes dex).

Having said the above, @tzabbi's issue is no longer a bug but a cleanup and refactoring.

@tzabbi , since you already started work in here, could you make the changes described above (remove oidc-autheservice definition and references)?

**Comment by juliusvonkohout** [comment]
> Hello all,
> 
> This wasn't an issue because the `example` kustomize correctly uses the `dex/overlays/oauth2-proxy` for dex.
> 
> Please see:
> 
> ```
> $ git log -1 
> commit ace875bbed9e284483c545cdc1c2cb60ad5b24a3 (HEAD -> upstream.master, upstream/master)
> Author:     biswajit-9776 <115724497+biswajit-9776@users.noreply.github.com>
> AuthorDate: Tue Jul 23 19:21:02 2024 +0530
> Commit:     GitHub <noreply@github.com>
> CommitDate: Tue Jul 23 13:51:02 2024 +0000
> 
>     PSS labels for the profile controller (#2778)
> 
> $ grep Dex -A1 example/kustomization.yaml 
> # Dex
> - ../common/dex/overlays/oauth2-proxy
> 
> $ kustomize build example 2>/dev/null | grep redirectURIs
>       redirectURIs: ["/oauth2/callback"]
> ```
> 
> But, I suppose with the changed default from the oidc-authservice to the oauth2-proxy, we can as well just delete the oidc-authservice from `common/oidc-client/oidc-authservice` (maybe even move `common/oidc-client/oauth2-proxy` to `common/oauth2-proxy`), and remove all references to the `oidc-authservice`, wherever they might be (this includes dex).
> 
> Having said the above, @tzabbi's issue is no longer a bug but a cleanup and refactoring.
> 
> @tzabbi , since you already started work in here, could you make the changes described above (remove oidc-autheservice definition and references)?

During the cleanup/refactoring we also have to address https://github.com/kubeflow/manifests/pull/2734#issuecomment-2145839700

https://github.com/kubeflow/manifests/blob/master/apps/centraldashboard/upstream/overlays/oauth2-proxy/kustomization.yaml is not allowed in the upstream folder. This breaks every time when we synchronize the upstream manifest. as mentioned here as well  https://github.com/kubeflow/manifests/issues/2804.

**Comment by tzabbi** [comment]
> Hello all,
> 
> This wasn't an issue because the `example` kustomize correctly uses the `dex/overlays/oauth2-proxy` for dex.
> 
> Please see:
> 
> ```
> $ git log -1 
> commit ace875bbed9e284483c545cdc1c2cb60ad5b24a3 (HEAD -> upstream.master, upstream/master)
> Author:     biswajit-9776 <115724497+biswajit-9776@users.noreply.github.com>
> AuthorDate: Tue Jul 23 19:21:02 2024 +0530
> Commit:     GitHub <noreply@github.com>
> CommitDate: Tue Jul 23 13:51:02 2024 +0000
> 
>     PSS labels for the profile controller (#2778)
> 
> $ grep Dex -A1 example/kustomization.yaml 
> # Dex
> - ../common/dex/overlays/oauth2-proxy
> 
> $ kustomize build example 2>/dev/null | grep redirectURIs
>       redirectURIs: ["/oauth2/callback"]
> ```
> 
> But, I suppose with the changed default from the oidc-authservice to the oauth2-proxy, we can as well just delete the oidc-authservice from `common/oidc-client/oidc-authservice` (maybe even move `common/oidc-client/oauth2-proxy` to `common/oauth2-proxy`), and remove all references to the `oidc-authservice`, wherever they might be (this includes dex).
> 
> Having said the above, @tzabbi's issue is no longer a bug but a cleanup and refactoring.
> 
> @tzabbi , since you already started work in here, could you make the changes described above (remove oidc-autheservice definition and references)?

@kromanow94 you are totally right. I cloned this repo again and get the same result. I might be the case that I modified this with a patch...

Yes, I will make this changes but not before tomorrow.

**Comment by tzabbi** [comment]
@juliusvonkohout it is still required to add a test? since my "bug" wasn't a bug.

File: README.md [hunk 1]
```diff
@@ -548,6 +548,6 @@ The Kubeflow security working group follows a responsible disclosure policy for
 ## Frequently Asked Questions
 
 - **Q:** What versions of Istio, Knative, Cert-Manager, Argo, ... are compatible with Kubeflow? \
-  **A:** Please refer to each individual component's documentation for a dependency compatibility range. For Istio, Knative, Dex, Cert-Manager and OIDC-AuthService, the versions in `common` are the ones we have validated.
+  **A:** Please refer to each individual component's documentation for a dependency compatibility range. For Istio, Knative, Dex, Cert-Manager and OAUTH2, the versions in `common` are the ones we have validated.
```
Comment (juliusvonkohout) [inline README.md:551]: Lets use the full name oauth2-proxy

File: apps/centraldashboard/upstream/base/deployment.yaml [hunk 2]
```diff
@@ -41,7 +41,7 @@ spec:
         - name: DASHBOARD_CONFIGMAP
           value: CD_CONFIGMAP_NAME_PLACEHOLDER
         - name: LOGOUT_URL
-          value: '/authservice/logout'
```
Comment (juliusvonkohout) [inline apps/centraldashboard/upstream/base/deployment.yaml:44]: This File is copied from upstream and must be changed in the kubeflow/kubeflow repository. We shall not modify upstream folders manually.

**Review by juliusvonkohout — CHANGES_REQUESTED** [review]

**Comment by juliusvonkohout** [comment]
> @juliusvonkohout it is still required to add a test? since my "bug" wasn't a bug.
> 

If you have the time yes. Mir e inportant night be 

"During the cleanup/refactoring we also have to address #2734 (comment)

https://github.com/kubeflow/manifests/blob/master/apps/centraldashboard/upstream/overlays/oauth2-proxy/kustomization.yaml is not allowed in the upstream folder. This breaks every time when we synchronize the upstream manifest. It is tracked here as well #2804."

File: apps/centraldashboard/upstream/overlays/oauth2-proxy/kustomization.yaml
Comment (juliusvonkohout) [inline apps/centraldashboard/upstream/overlays/oauth2-proxy/kustomization.yaml:1]: During the cleanup/refactoring we also have to address https://github.com/kubeflow/manifests/pull/2734#issuecomment-2145839700

https://github.com/kubeflow/manifests/blob/master/apps/centraldashboard/upstream/overlays/oauth2-proxy/kustomization.yaml is not allowed in the upstream folder. This breaks every time when we synchronize the upstream manifest. It is tracked here as well https://github.com/kubeflow/manifests/issues/2804.

File: apps/centraldashboard/manuel-patches/oauth2-proxy/kustomization.yaml
Comment (juliusvonkohout) [inline apps/centraldashboard/manuel-patches/oauth2-proxy/kustomization.yaml:1]: i think it should be manuell not manuell

File: example/kustomization.yaml [hunk 3]
```diff
@@ -64,7 +64,7 @@ resources:
 # Katib
 - ../apps/katib/upstream/installs/katib-with-kubeflow
 # Central Dashboard
-- ../apps/centraldashboard/upstream/overlays/oauth2-proxy
+- ../apps/centraldashboard/manuel-patches/oauth2-proxy
```
Comment (juliusvonkohout) [inline example/kustomization.yaml:67]: Probably we should call it just overlays instead of manuell-patches

Comment (tzabbi) [reply example/kustomization.yaml:67]: this is fine for me

File: common/oidc-client/README.md
Comment (juliusvonkohout) [inline common/oidc-client/README.md:1]: @kromanow94 are we sure that all this information is in the other readme? @tzabbi  I think we have to merge them properly

Comment (juliusvonkohout) [reply common/oidc-client/README.md:1]: Especially the section ## Kubeflow Pipelines User and M2M Authentication and Authorization and below

Comment (juliusvonkohout) [reply common/oidc-client/README.md:1]: You can also just leave it in for now @tzabbi if it is too much effort to merge it.

**Comment by juliusvonkohout** [comment]
Regarding

```
Run kustomize build example
  kustomize build example
  shell: /usr/bin/bash -e {0}
# Warning: 'vars' is deprecated. Please use 'replacements' instead. [EXPERIMENTAL] Run 'kustomize edit fix' to update your Kustomization automatically.
# Warning: 'patchesStrategicMerge' is deprecated. Please use 'patches' instead. Run 'kustomize edit fix' to update your Kustomization automatically.
# Warning: 'vars' is deprecated. Please use 'replacements' instead. [EXPERIMENTAL] Run 'kustomize edit fix' to update your Kustomization automatically.
# Warning: 'patchesStrategicMerge' is deprecated. Please use 'patches' instead. Run 'kustomize edit fix' to update your Kustomization automatically.
# Warning: 'vars' is deprecated. Please use 'replacements' instead. [EXPERIMENTAL] Run 'kustomize edit fix' to update your Kustomization automatically.
Error: accumulating resources: accumulation err='accumulating resources from '../apps/centraldashboard/overlays': '/home/runner/work/manifests/manifests/apps/centraldashboard/overlays' must resolve to a file': couldn't make target for path '/home/runner/work/manifests/manifests/apps/centraldashboard/overlays': unable to find one of 'kustomization.yaml', 'kustomization.yml' or 'Kustomization' in directory '/home/runner/work/manifests/manifests/apps/centraldashboard/overlays'
```

i think it should be ../apps/centraldashboard/overlays/oauth2-proxy in /example/kustomization.yaml

**Comment by tzabbi** [comment]
im on it. @juliusvonkohout I can mark it as WIP if you want. So you havn't take a look all the time.

File: common/oidc-client/README.md
Comment (tzabbi) [reply common/oidc-client/README.md:1]: Added this content again, but I'm not sure how to adjust it properly.

**Comment by tzabbi** [comment]
@juliusvonkohout fixed all of our comments :)

**Comment by juliusvonkohout** [comment]
@kromanow94 i think it is ready for your review. I will handle the formatting of deleted files later on.

**Comment by juliusvonkohout** [comment]
Alright now deleted files are ignored, but we have to fix

```
.github/workflows/dex_test.yaml
  Warning: 2:1 [truthy] truthy value should be one of [false, true]
  Error: 5:7 [indentation] wrong indentation: expected 4 but found 6
  Error: 38:12 syntax error: mapping values are not allowed here (syntax)

.github/workflows/kserve_m2m_test.yaml
  Warning: 2:1 [truthy] truthy value should be one of [false, true]
  Error: 5:7 [indentation] wrong indentation: expected 4 but found 6
  Error: 23:7 [indentation] wrong indentation: expected 4 but found 6

.github/workflows/notebook_controller_m2m_test.yaml
  Warning: 2:1 [truthy] truthy value should be one of [false, true]
  Error: 5:7 [indentation] wrong indentation: expected 4 but found 6

.github/workflows/pipeline_run_from_notebook.yaml
  Warning: 2:1 [truthy] truthy value should be one of [false, true]
  Error: 5:7 [indentation] wrong indentation: expected 4 but found 6

.github/workflows/pipeline_test.yaml
  Warning: 2:1 [truthy] truthy value should be one of [false, true]
  Error: 5:7 [indentation] wrong indentation: expected 4 but found 6
  Error: 21:7 [indentation] wrong indentation: expected 4 but found 6

Error: Process completed with exit code 1.
```

**Comment by tzabbi** [comment]
im on it.

**Comment by tzabbi** [comment]
@juliusvonkohout what is with the yaml linter should i fix it?

**Comment by hansinikarunarathne** [comment]
> @hansinikarunarathne it is better now, but still not 100% fixed I think

I think @tzabbi  has not taken the changes I made to his branch. Can you check it @tzabbi ?

**Comment by tzabbi** [comment]
Hi @hansinikarunarathne which change do you mean? were can i found your changes?

**Comment by hansinikarunarathne** [comment]
> Hi @hansinikarunarathne which change do you mean? were can I find your changes?

Hi @tzabbi. I did the fix of trying to linting deleted YAML files in this PR #2823. Please feel free to check my changes here.  Changes are in the master. You can take it from the master. If you further face issues regarding this, let me know

**Comment by tzabbi** [comment]
@hansinikarunarathne on master? I already pulled it :)

**Comment by hansinikarunarathne** [comment]
> @hansinikarunarathne on master? I already pulled it :)

In your master, I see this 
image

But I changed it into 
image

So that's why I mentioned that your branch is not up to master of Kubelow manifest. If I am wrong , make me correct

**Comment by tzabbi** [comment]
Looks like you are right. I will check the inconsistency tomorrow.

**Comment by tzabbi** [comment]
Added your changes @hansinikarunarathne can you confirm that all your changes are now in the PR?

**Comment by juliusvonkohout** [comment]
It is still quite similar 
```
2024-07-31T09:28:39.8665544Z ##[group].github/workflows/admission_webhook_test.yaml
2024-07-31T09:28:39.8695531Z ##[warning]2:1 [truthy] truthy value should be one of [false, true]
2024-07-31T09:28:39.8702846Z ##[endgroup]
2024-07-31T09:28:39.8703184Z 
2024-07-31T09:28:39.9363533Z ##[group].github/workflows/bentoml_test.yaml
2024-07-31T09:28:39.9365864Z ##[warning]2:1 [truthy] truthy value should be one of [false, true]
2024-07-31T09:28:39.9367633Z ##[endgroup]
2024-07-31T09:28:39.9367936Z 
2024-07-31T09:28:40.0055855Z ##[group].github/workflows/centraldashboard_test.yaml
2024-07-31T09:28:40.0057599Z ##[warning]2:1 [truthy] truthy value should be one of [false, true]
2024-07-31T09:28:40.0059247Z ##[endgroup]
2024-07-31T09:28:40.0059443Z 
2024-07-31T09:28:40.0763924Z ##[group].github/workflows/dex_test.yaml
2024-07-31T09:28:40.0766359Z ##[warning]2:1 [truthy] truthy value should be one of [false, true]
2024-07-31T09:28:40.0772053Z ##[error]46:7 syntax error: could not find expected ':' (syntax)
2024-07-31T09:28:40.0774426Z ##[endgroup]
2024-07-31T09:28:40.0774735Z 
2024-07-31T09:28:40.1472323Z ##[group].github/workflows/jupyter_web_application_test.yaml
2024-07-31T09:28:40.1474979Z ##[warning]2:1 [truthy] truthy value should be one of [false, true]
2024-07-31T09:28:40.1477600Z ##[endgroup]
2024-07-31T09:28:40.1477899Z 
2024-07-31T09:28:40.2191354Z ##[group].github/workflows/katib_test.yaml
2024-07-31T09:28:40.2194291Z ##[warning]2:1 [truthy] truthy value should be one of [false, true]
2024-07-31T09:28:40.2196770Z ##[endgroup]
2024-07-31T09:28:40.2197113Z 
2024-07-31T09:28:40.2942298Z ##[group].github/workflows/kserve_cni_test.yaml
2024-07-31T09:28:40.2944726Z ##[warning]2:1 [truthy] truthy value should be one of [false, true]
2024-07-31T09:28:40.2947252Z ##[endgroup]
2024-07-31T09:28:40.2947566Z 
2024-07-31T09:28:40.3722192Z ##[group].github/workflows/kserve_m2m_test.yaml
2024-07-31T09:28:40.3724141Z ##[warning]2:1 [truthy] truthy value should be one of [false, true]
2024-07-31T09:28:40.3726154Z ##[endgroup]
2024-07-31T09:28:40.3726368Z 
2024-07-31T09:28:40.4478913Z ##[group].github/workflows/kserve_test.yaml
2024-07-31T09:28:40.4480632Z ##[warning]2:1 [truthy] truthy value should be one of [false, true]
2024-07-31T09:28:40.4482315Z ##[endgroup]
2024-07-31T09:28:40.4482558Z 
2024-07-31T09:28:40.5292891Z ##[group].github/workflows/linting_bash_python_yaml_files.yaml
2024-07-31T09:28:40.5294958Z ##[warning]3:1 [truthy] truthy value should be one of [false, true]
2024-07-31T09:28:40.5297725Z ##[warning]128:20 [comments] too few spaces before comment
2024-07-31T09:28:40.5299532Z ##[endgroup]
2024-07-31T09:28:40.5299741Z 
2024-07-31T09:28:40.5958687Z ##[group].github/workflows/manifests_example_test.yaml
2024-07-31T09:28:40.5961196Z ##[warning]3:1 [truthy] truthy value should be one of [false, true]
2024-07-31T09:28:40.5964821Z ##[error]22:1 [empty-lines] too many blank lines (1 > 0)
2024-07-31T09:28:40.5967084Z ##[endgroup]
2024-07-31T09:28:40.5967378Z 
2024-07-31T09:28:40.6748942Z ##[group].github/workflows/metacontroller_test.yaml
2024-07-31T09:28:40.6751442Z ##[warning]2:1 [truthy] truthy value should be one of [false, true]
2024-07-31T09:28:40.6753951Z ##[endgroup]
2024-07-31T09:28:40.6754274Z 
2024-07-31T09:28:40.7517567Z ##[group].github/workflows/model_registry_test.yaml
2024-07-31T09:28:40.7519995Z ##[error]1:78 [trailing-spaces] trailing spaces
2024-07-31T09:28:40.7523664Z ##[warning]4:1 [truthy] truthy value should be one of [false, true]
2024-07-31T09:28:40.7526932Z ##[error]64:145 [trailing-spaces] trailing spaces
2024-07-31T09:28:40.7528979Z ##[endgroup]
2024-07-31T09:28:40.7529243Z 
2024-07-31T09:28:40.8304637Z ##[group].github/workflows/notebook_controller_m2m_test.yaml
2024-07-31T09:28:40.8307106Z ##[warning]2:1 [truthy] truthy value should be one of [false, true]
2024-07-31T09:28:40.8309588Z ##[endgroup]
2024-07-31T09:28:40.8309944Z 
2024-07-31T09:28:40.9007695Z ##[group].github/workflows/notebook_controller_test.yaml
2024-07-31T09:28:40.9010218Z ##[warning]2:1 [truthy] truthy value should be one of [false, true]
2024-07-31T09:28:40.9012951Z ##[endgroup]
2024-07-31T09:28:40.9013286Z 
2024-07-31T09:28:40.9777687Z ##[group].github/workflows/pipeline_run_from_notebook.yaml
2024-07-31T09:28:40.9779977Z ##[warning]2:1 [truthy] truthy value should be one of [false, true]
2024-07-31T09:28:40.9782364Z ##[endgroup]
2024-07-31T09:28:40.9782725Z 
2024-07-31T09:28:41.0560419Z ##[group].github/workflows/pipeline_test.yaml
2024-07-31T09:28:41.0562846Z ##[warning]2:1 [truthy] truthy value should be one of [false, true]
2024-07-31T09:28:41.0565424Z ##[endgroup]
2024-07-31T09:28:41.0565724Z 
2024-07-31T09:28:41.1266392Z ##[group].github/workflows/profiles_test.yaml
2024-07-31T09:28:41.1267964Z ##[warning]2:1 [truthy] truthy value should be one of [false, true]
2024-07-31T09:28:41.1269542Z ##[endgroup]
2024-07-31T09:28:41.1269731Z 
2024-07-31T09:28:41.1964129Z ##[group].github/workflows/ray_test.yaml
2024-07-31T09:28:41.1966223Z ##[warning]2:1 [truthy] truthy value should be one of [false, true]
2024-07-31T09:28:41.1968237Z ##[endgroup]
2024-07-31T09:28:41.1968491Z 
2024-07-31T09:28:41.2675013Z ##[group].github/workflows/seldon_test.yaml
2024-07-31T09:28:41.2677098Z ##[warning]2:1 [truthy] truthy value should be one of [false, true]
2024-07-31T09:28:41.2679292Z ##[endgroup]
2024-07-31T09:28:41.3405243Z 
2024-07-31T09:28:41.3406400Z ##[group].github/workflows/stale.yaml
2024-07-31T09:28:41.3408930Z ##[warning]8:1 [truthy] truthy value should be one of [false, true]
2024-07-31T09:28:41.3412222Z ##[warning]10:23 [comments] too few spaces before comment
2024-07-31T09:28:41.3414851Z ##[error]25:68 [trailing-spaces] trailing spaces
2024-07-31T09:28:41.3416955Z ##[error]29:68 [trailing-spaces] trailing spaces
2024-07-31T09:28:41.3419053Z ##[error]36:75 [trailing-spaces] trailing spaces
2024-07-31T09:28:41.3422058Z ##[error]39:75 [trailing-spaces] trailing spaces
2024-07-31T09:28:41.3424322Z ##[endgroup]
2024-07-31T09:28:41.3424527Z 
2024-07-31T09:28:41.4122231Z ##[group].github/workflows/tensorboard_controller_test.yaml
2024-07-31T09:28:41.4124831Z ##[warning]2:1 [truthy] truthy value should be one of [false, true]
2024-07-31T09:28:41.4127316Z ##[endgroup]
2024-07-31T09:28:41.4127671Z 
2024-07-31T09:28:41.4853419Z ##[group].github/workflows/tensorboards_web_application_test.yaml
2024-07-31T09:28:41.4857158Z ##[warning]2:1 [truthy] truthy value should be one of [false, true]
2024-07-31T09:28:41.4859722Z ##[endgroup]
2024-07-31T09:28:41.4860033Z 
2024-07-31T09:28:41.5629083Z ##[group].github/workflows/training_operator_test.yaml
2024-07-31T09:28:41.5631627Z ##[warning]2:1 [truthy] truthy value should be one of [false, true]
2024-07-31T09:28:41.5634070Z ##[endgroup]
2024-07-31T09:28:41.5634452Z 
2024-07-31T09:28:41.6326632Z ##[group].github/workflows/triage_issues.yaml
2024-07-31T09:28:41.6328909Z ##[error]1:62 [trailing-spaces] trailing spaces
2024-07-31T09:28:41.6331832Z ##[warning]4:1 [truthy] truthy value should be one of [false, true]
2024-07-31T09:28:41.6333953Z ##[error]19:1 [empty-lines] too many blank lines (1 > 0)
2024-07-31T09:28:41.6335343Z ##[endgroup]
2024-07-31T09:28:41.6335526Z 
2024-07-31T09:28:41.7041944Z ##[group].github/workflows/trivy.yaml
2024-07-31T09:28:41.7043988Z ##[warning]3:1 [truthy] truthy value should be one of [false, true]
2024-07-31T09:28:41.7046958Z ##[error]50:1 [empty-lines] too many blank lines (2 > 0)
2024-07-31T09:28:41.7048694Z ##[endgroup]
2024-07-31T09:28:41.7049310Z 
2024-07-31T09:28:41.7753260Z ##[group].github/workflows/volumes_web_application_test.yaml
2024-07-31T09:28:41.7756068Z ##[warning]2:1 [truthy] truthy value should be one of [false, true]
2024-07-31T09:28:41.7759283Z ##[endgroup]
2024-07-31T09:28:41.7759649Z 
2024-07-31T09:28:42.4236188Z ##[error]Process completed with exit code 1.
```

at least the errors have to be fixed.

**Comment by juliusvonkohout** [comment]
But please do not make this PR bigger than it needs to be. Lets only fix files that have to changed here right now.

**Comment by tzabbi** [comment]
Sorry for the whole syntax missmatches i have to check my linter...

**Comment by tzabbi** [comment]
Fixed all errors.

**Comment by juliusvonkohout** [comment]
I see 

```
Run pip3 install requests
  pip3 install requests
  ./tests/gh-actions/test_dex_login.py
  shell: /usr/bin/bash -e {0}
Defaulting to user installation because normal site-packages is not writeable
Requirement already satisfied: requests in /usr/lib/python3/dist-packages (2.25.1)
/home/runner/work/_temp/0cb81e68-c08c-4a89-b642-a24c5f9eb9a3.sh: line 2: ./tests/gh-actions/test_dex_login.py: Permission denied
Error: Process completed with exit code 126.
```

Maybe a chmod +x is needed.

File: common/oauth2-proxy/components/README.md [hunk 4]
```diff
@@ -208,10 +208,10 @@ index c1a85789..4a50440c 100644
  - ../common/istio-1-22/istio-namespace/base
 -- ../common/istio-1-22/istio-install/base
 -# OIDC Authservice
--- ../common/oidc-client/oidc-authservice/base
+-- ../common//oidc-authservice/base
```
Comment (juliusvonkohout) [inline common/oauth2-proxy/components/README.md:211]: i think oidc-authservice should be removed including the comment

**Comment by juliusvonkohout** [comment]
In general
/lgtm
/approve

as of now, but please fix the dex test only and nothing else, or i have to review 100 files from scratch again.

**Comment by juliusvonkohout** [comment]
Let us continue in https://github.com/kubeflow/manifests/pull/2827

**Pull request merged**